### PR TITLE
move defer before the creation

### PR DESCRIPTION
**Pull request by Xia-Zhao-rh**

### CODE REVIEW

File: test/extended/operators/olm.go [hunk 1]
```diff
@@ -597,19 +597,19 @@ var _ = g.Describe("[sig-operators] OLM for an end user handle within all namesp
...
 		defer subElasticSearch.delete(itName, dr)
+		subElasticSearch.create(oc, itName, dr)
+		csvElasticSearch.name = subElasticSearch.installedCSV
 		newCheck("expect", asAdmin, withoutNamespace, compare, "Succeeded", ok, []string{"csv", subElasticSearch.installedCSV, "-n", subElasticSearch.namespace, "-o=jsonpath={.status.phase}"}).check(oc)
 
 		g.By("create operator Jaeger")
-		subJaeger.create(oc, itName, dr)
-		csvJaeger.name = subJaeger.installedCSV
 		defer crdJaegers.delete(oc)
 		defer csvJaeger.delete(itName, dr)
```
Comment (kuiwang02) [inline test/extended/operators/olm.go]: @Xia-Zhao-rh please move defer csvJaeger.delete(itName, dr) after csvJaeger.name = subJaeger.installedCSV
acutaully derfer csv could be after sub.create. for this case, the csv.name is "" originally, so please move it after that line.

same comment for defer csvElasticSearch.delete(itName, dr)

**Comment by jianzhangbjz** [comment]
@Xia-Zhao-rh There is a conflict here.

**Comment by Xia-Zhao-rh** [comment]
the conflict has been fixed.

**Pull request merged**